File: src/_03_BehavioralPattern/_03_01_ChainOfResponsibilities/after/AuthRequestHandler.java
package _03_BehavioralPattern._03_01_ChainOfResponsibilities.after;

public class AuthRequestHandler extends RequestHandler {

  public AuthRequestHandler(RequestHandler nextHandler) {
    super(nextHandler);
  }

  @Override
  public void handler(Request request) {
    System.out.println("인증 확인 완료");
    super.handler(request);
  }
  
}

File: src/_03_BehavioralPattern/_03_01_ChainOfResponsibilities/AuthRequestHandler.js
import RequestHandler from "./RequestHandler.js";

/**
 * @typedef { import("./Request").default } Request
 */

export default class AuthRequestHandler extends RequestHandler {
  /** @param { RequestHandler } nextHandler */
  constructor(nextHandler) {
    super(nextHandler);
  }

  /**
   * @override
   * @param { Request } request 
   */
  handler(request) {
    console.log("인증 되었습니다.");
    super.handler(request);
  }
}